src/riverroute/mosart_restfile.F90

```diff
@@ -0,0 +1,455 @@
+
+   character(len=256) function mosart_rest_FileName( rdate )
+
+      ! Arguments
+      character(len=*), intent(in) :: rdate   ! input date for restart file name
+
+      mosart_rest_FileName = "./"//trim(caseid)//".mosart"//trim(inst_suffix)//".r."//trim(rdate)//".nc"
+      if (mainproc) then
+         write(iulog,*)'writing restart file ',trim(mosart_rest_FileName),' for model date = ',rdate
+      end if
+
+   end function mosart_rest_FileName
+
+   !------------------------------------------------------------------------
+
+   subroutine restFile_dimset( ncid )
+
+      !-------------------------------------
+      ! Read/Write initial data from/to netCDF instantaneous initial data file
+
+      ! Arguments
+      type(file_desc_t), intent(inout) :: ncid
+
+      ! Local Variables:
+      integer :: dimid               ! netCDF dimension id
+      integer :: ier                 ! error status
+      character(len=  8) :: curdate  ! current date
+      character(len=  8) :: curtime  ! current time
+      character(len=256) :: str
+      character(len=*),parameter :: subname='restFile_dimset'
+      !-------------------------------------
+
+      ! Define dimensions
+
+      call ncd_defdim(ncid, 'nlon'  , ctl%nlon  , dimid)
+      call ncd_defdim(ncid, 'nlat'  , ctl%nlat  , dimid)
+      call ncd_defdim(ncid, 'string_length', 64 , dimid)
+
+      ! Define global attributes
+
+      call ncd_putatt(ncid, NCD_GLOBAL, 'Conventions', trim(conventions))
+      call getdatetime(curdate, curtime)
+      str = 'created on ' // curdate // ' ' // curtime
+      call ncd_putatt(ncid, NCD_GLOBAL, 'history' , trim(str))
+      call ncd_putatt(ncid, NCD_GLOBAL, 'username', trim(username))
+      call ncd_putatt(ncid, NCD_GLOBAL, 'host'    , trim(hostname))
+      call ncd_putatt(ncid, NCD_GLOBAL, 'version' , trim(version))
+      call ncd_putatt(ncid, NCD_GLOBAL, 'source'  , trim(source))
+      call ncd_putatt(ncid, NCD_GLOBAL, 'case_title'     , trim(ctitle))
+      call ncd_putatt(ncid, NCD_GLOBAL, 'case_id'        , trim(caseid))
+      call ncd_putatt(ncid, NCD_GLOBAL, 'title', &
+           'MOSART Restart information, required to continue a simulation' )
+
+   end subroutine restFile_dimset
+
+   !-----------------------------------------------------------------------
+
+   subroutine mosart_rest_restart(ncid, flag)
+
+      !-------------------------------------
+      ! Read/write MOSART restart data.
+      !
+      ! Arguments:
+      type(file_desc_t), intent(inout)  :: ncid ! netcdf id
+      character(len=*) , intent(in) :: flag   ! 'read' or 'write'
+
+      ! Local variables
+      logical :: readvar          ! determine if variable is on initial file
+      integer :: nt,nv,n          ! indices
+      real(r8) , pointer :: dfld(:) ! temporary array
+      character(len=32)  :: vname,uname
+      character(len=255) :: lname
+      !-------------------------------------
+
+      do nv = 1,7
```

Comment:
Remove hard coded values, use descriptive parameters.

Reply:
Done.

Reply:
not done?